src/PrestaShopBundle/DependencyInjection/Compiler/LoadDoctrineFromModulesPassFactory.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\DependencyInjection\Compiler;
+
+use Doctrine\Bundle\DoctrineBundle\DependencyInjection\Compiler\DoctrineOrmMappingsPass;
+use Doctrine\Common\Util\Inflector;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Finder\SplFileInfo;
+
+/**
+ * Load services stored in installed modules.
+ */
+class LoadDoctrineFromModulesPassFactory
+{
+    /**
+     * @param array $installedModules
+     *
+     * @return CompilerPassInterface[]
+     */
+    public function buildCompilerPassList(array $installedModules)
+    {
+        return $this->buildAnnotationMappingsPass($installedModules);
+    }
+
+    private function buildAnnotationMappingsPass(array $installedModules)
+    {
+        $mappingsPassList = [];
+        /** @var SplFileInfo $modulePath */
+        foreach ($this->getModulesPaths() as $modulePath) {
+            if (in_array($modulePath->getFilename(), $installedModules)
+                && file_exists($modulePath . '/src/Entity')
```

Comment:
I see you take a "zero configuration" approach and decided to add a `DoctrineOrmMappingsPass` if `src/Entity` exists.

It should probably be possible to look for files ending with '.orm.xml' or 'orm.yml' inside a specific folder (like config/mappings or whatever) and depending of what is found or not, create the approproate compiler pass.

Does that sound like something you could agree on ?

Reply:
yes, I could include what you did in your PR, actually that's why I separated this behaviour in a `buildAnnotationMappingsPass` method
let me confirm it with the team before

Reply:
I'm not sure we should allow xml/yml file support here: tbh I hate that people needs to learn 2, 3 or 4 ways to do the same stuff. Don't forget that if you need to work with PrestaShop the "Symfony way", the best is to create and register your own bundle!

Now, about PrestaShop developers: I think it's more "developer friendly" to have only 1 way to do it. Except if I'm wrong, Doctrine is dropping the support of Yaml, and most of developers I know hate XML files: this is why I'm in favor of the annotations.

> And once we will find a way to enable the `Route` annotation in module controllers and service autowiring, PrestaShop developers will almost never touch a Yaml file neither :+1:

Reply:
Hi,
Thanks, I'll keep the advice for a bundle in mind. In my current case, I was toying with the idea of writing a public module and see how far I could keep my symfony habits, which brought me here :-)
Annotations will be fine for me (I was just thinking that separate config would have been useful for module authors wanting to use models coming from separate packages)